platform/qt/src/timer.cpp

```diff
@@ -18,7 +18,7 @@ void Timer::Impl::start(uint64_t timeout, uint64_t repeat_, std::function<void (
     callback = std::move(cb);
 
     timer.setSingleShot(true);
-    timer.start(timeout);
+    timer.start(static_cast<std::chrono::milliseconds>(timeout));
```

Comment:
This line fails on SFOS with error 

```
[ 1011s] /home/abuild/rpmbuild/BUILD/qmaplibregl-0.0.220220.3+ntadej.qt.improvements.20220428192425.1.gb8426c6/maplibre-gl-native/platform/qt/src/timer.cpp:21:64: error: no matching function for call to 'QTimer::start(std::chrono::milliseconds)'
[ 1011s]      timer.start(static_cast<std::chrono::milliseconds>(timeout));
[ 1011s]                                                                 ^
[ 1011s] In file included from /usr/include/qt5/QtCore/QTimer:1,
[ 1011s]                  from /home/abuild/rpmbuild/BUILD/qmaplibregl-0.0.220220.3+ntadej.qt.improvements.20220428192425.1.gb8426c6/maplibre-gl-native/platform/qt/src/timer_impl.hpp:6,
[ 1011s]                  from /home/abuild/rpmbuild/BUILD/qmaplibregl-0.0.220220.3+ntadej.qt.improvements.20220428192425.1.gb8426c6/maplibre-gl-native/platform/qt/src/timer.cpp:1:
[ 1011s] /usr/include/qt5/QtCore/qtimer.h:142:10: note: candidate: 'void QTimer::start(int)'
[ 1011s]      void start(int msec);
[ 1011s]           ^~~~~
[ 1011s] /usr/include/qt5/QtCore/qtimer.h:142:10: note:   no known conversion for argument 1 from 'std::chrono::milliseconds' {aka 'std::chrono::duration<long long int, std::ratio<1, 1000> >'} to 'int'
[ 1011s] /usr/include/qt5/QtCore/qtimer.h:144:10: note: candidate: 'void QTimer::start()'
[ 1011s]      void start();
[ 1011s]           ^~~~~
[ 1011s] /usr/include/qt5/QtCore/qtimer.h:144:10: note:   candidate expects 0 arguments, 1 provided
```

Same goes for a line 31 below. 

What is this conversion designed to achieve? Isn't timeout in ms already?

Reply:
As a comment: we are on Qt 5.6 due to license changes in the latter Qt releases. So far, it hasn't been an issue for Mapbox/Maplibre Qt Native

Reply:
OK, so we can set Qt 5.6 and CMake 3.10 as minimum supported for now. Will setup a CI pipeline.

Reply:
It seems Qt changed the interface from `int` to `std::chronology::milliseconds` in version 5.8, fixed.